src/app/Fake.Core.Target/Target.fs

```diff
@@ -584,15 +596,16 @@ module Target =
                 member __.GetNextTarget (ctx) = mbox.PostAndAsyncReply(fun reply -> GetNextTarget(ctx, reply))
             }
 
-        let runOptimal workerNum (order:Target[] list) targetContext =
+        let runOptimal workerNum (order:Target[] list) targetContext=
             let mgr = createCtxMgr order targetContext
             let targetRunner () =
                 async {
+                    let token = targetContext.CancellationToken
```

Comment:
Instead of using the token from the context here, should i add it to `Async.StartAsTask` and `Async.RunSynchronously` ? I guess i would have to add it to both because of the Task.AwaitAll?

Reply:
Actually I think it is fine as it is, using the default cancellation token has funny side effects. We want the runner to stay alive in order to execute the final targets.